plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts

```diff
@@ -99,6 +102,17 @@ export class GithubPublisher implements PublisherBase {
       /\.git$/,
       '/blob/master/catalog-info.yaml',
     );
+
+    try {
+      await enableBranchProtectionOnDefaultRepoBranch({
```

Comment:
Just realized this is the deprecated GitHub publisher, we should add it to `plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts` instead. Could prolly add it to both though.

Reply:
How is it deprecated if it works though? See the GIF in original PR description.

Reply:
@benjdlambert is on top of moving things around atm 😁

Deprecated just means that we're phasing it out. In this case it's the `alpha1` templates being replaced by the `beta2` ones, so if you're using an old template it will be executing this code.

Reply:
Awesome! Thanks! So – if I understand correctly – I can test it by using a newer template.

Reply:
Yep - you can try the `Test Action template` template instead. That's the later action format. 👍

Reply:
Awesome! Thanks, Ben!